core/src/main/java/org/apache/iceberg/view/ViewVersionReplace.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import static org.apache.iceberg.TableProperties.COMMIT_MAX_RETRY_WAIT_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_MAX_RETRY_WAIT_MS_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_MIN_RETRY_WAIT_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_MIN_RETRY_WAIT_MS_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_NUM_RETRIES;
+import static org.apache.iceberg.TableProperties.COMMIT_NUM_RETRIES_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_TOTAL_RETRY_TIME_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_TOTAL_RETRY_TIME_MS_DEFAULT;
+
+import java.util.List;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.util.PropertyUtil;
+import org.apache.iceberg.util.Tasks;
+
+class ViewVersionReplace implements ReplaceViewVersion {
+  private final ViewOperations ops;
+  private final List<ViewRepresentation> viewRepresentationsToAdd = Lists.newArrayList();
+  private ViewMetadata base;
+  private Namespace defaultNamespace;
+  private String defaultCatalog;
+  private Schema schema;
+
+  ViewVersionReplace(ViewOperations ops) {
+    this.ops = ops;
+    this.base = ops.current();
+  }
+
+  @Override
+  public ViewVersion apply() {
+    Preconditions.checkState(
+        !viewRepresentationsToAdd.isEmpty(), "Cannot replace view without specifying a query");
+
+    this.base = ops.refresh();
+
+    ViewVersion viewVersion = base.currentVersion();
+
+    return ImmutableViewVersion.builder()
+        .versionId(viewVersion.versionId() + 1)
+        .timestampMillis(System.currentTimeMillis())
+        .schemaId(null == schema ? viewVersion.schemaId() : schema.schemaId())
+        .defaultNamespace(defaultNamespace)
```

Comment:
What validates that this is non-null?

Reply:
this is required to be non-null as defined in the `ViewVersion` API: https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/view/ViewVersion.java#L81

So it's not possible to create an instance where this ever ends up being null.

Reply:
@nastra thanks for the link, I see that it's enforced by the Immutables builder. I think there should also be a check here. It is a required field and the Immutables check is not something that we should rely on.

I also noticed that there's a `check` method in the `ViewVersion` interface in API at that link. Looks like that was an oversight -- it's not part of the API so it should be removed along with any others that were accidentally included before the recent refactor. This is a high priority since we are making breaking changes in the 1.4 release and we want to get them all done.